iterator/combined_iterator.go

```diff
@@ -0,0 +1,151 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iterator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodbstreams"
+	"github.com/conduitio-labs/conduit-connector-dynamodb/position"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type CombinedIterator struct {
+	snapshotIterator *SnapshotIterator
+	cdcIterator      *CDCIterator
+
+	tableName     string
+	partitionKey  string
+	sortKey       string
+	client        *dynamodb.Client
+	streamsClient *dynamodbstreams.Client
+	streamArn     string
+	pollingPeriod time.Duration
+}
+
+func NewCombinedIterator(
+	ctx context.Context,
+	tableName, pKey, sKey string,
+	pollingPeriod time.Duration,
+	client *dynamodb.Client,
+	streamsClient *dynamodbstreams.Client,
+	streamArn string,
+	p position.Position,
+) (*CombinedIterator, error) {
+	var err error
+	c := &CombinedIterator{
+		tableName:     tableName,
+		partitionKey:  pKey,
+		sortKey:       sKey,
+		pollingPeriod: pollingPeriod,
+		client:        client,
+		streamsClient: streamsClient,
+		streamArn:     streamArn,
+	}
+
+	switch p.Type {
+	case position.TypeSnapshot:
+		if len(p.Key) != 0 {
+			sdk.Logger(ctx).
+				Warn().
+				Str("position", string(p.ToRecordPosition())).
+				Msg("previous snapshot did not complete successfully. snapshot will be restarted for consistency.")
```

Comment:
yes that's the goal, but couldn't find any data about what will happen if a record is added while I'm doing the snapshot, is the snapshot gonna get it too? (cuz I'm getting the snapshot page by page), which makes it a duplication of the record when CDC is on. 
will have to do more research here and maybe some manual tests to figure it out, this also was a quick solution for demo purposes, since the `S3` connector does the same thing to prevent dups

Reply:
yes that's the goal, but couldn't find any data about what will happen if a record is added while I'm doing the snapshot, is the snapshot gonna get it too? (cuz I'm getting the snapshot page by page), which makes it a duplication of the record when CDC is on. 
will have to do more research here and maybe some manual tests to figure it out, this also was a quick solution for demo purposes, since the `S3` connector does the same thing to prevent dups

Reply:
ok did more research, looks like it has the same behavior as the S3 snapshot, once the snapshot scanning starts, any updates while scanning will not be scanned, which means if the pipeline restarts and we try to continue the snapshot, a certain record could be updated from the records already scanned and we wouldn't know.

Reply:
Hmm, that's unfortunate... Can we start the CDC iterator at the same time, so that it's listening to those changes during the snapshot?

Reply:
yep that's in the new changes, it's just not pushed yet.